Music service

bsutton opened the pull request:
Added logic to parse the MusicServices into a java model.

bsutton committed:
added 'def' to service variable so it will compile

bsutton committed:
Added constant for speaker max volumen.

bsutton committed:
Added toString so that when debugging the device's friendly name is
displayed.

bsutton committed:
Add comments to the volume related methods.

bsutton committed:
Added a model for MusicService and a parser to extract the model from
the xml returned by the MusicService service.

bsutton committed:
Made the PresentationMap optional.

rattigan commented:
Any chance you could switch your editor to use 4 spaces instead of tabs? It's hard to read the pr right now as everything shows up as changed.

I haven't looked at this project in quite a while - are you finding it useful?

bsutton committed:
Added a experimental BrowserType to make the browse options clearer

bsutton committed:
Added a class to translate sonos pnp error codes to useful error
strings.

bsutton committed:
Change parent exception from RuntimeException to Exception to force
library users to catch the exception.

Created experiemental BrowseType enum.

bsutton committed:
Made the sonos exception inherit from Exception rather than
RuntimeException so that library users would correctly catch and handle
exceptions rather then ignoring them.

bsutton committed:
Corrected a typo in a comment.

bsutton committed:
Added getters to some fields.

bsutton committed:
improved a comment.

bsutton committed:
Added getters.

bsutton committed:
Made the sonos exception inherit from Exception rather than
RuntimeException so that library users would correctly catch and handle
exceptions rather then ignoring them.

bsutton committed:
Made the sonos exception inherit from Exception rather than
RuntimeException so that library users would correctly catch and handle
exceptions rather then ignoring them.

Changed code to use the new BrowseType.

bsutton commented:
I've sent a new pull request with the correct spacing.

As to useful, I'm still experimenting. The hard part is trying to get my 
head around what can actually be done via the api.
Originally I assume that I would be able to query music services and 
display the list of music available on the service (just as the sonos 
app can) but this doesn't look like the case.

My reading is that sonos are considering publishing an api but still 
haven't. Am I correct?

brett

On 31/12/15 11:33, rattigan wrote:

> Any chance you could switch your editor to use 4 spaces instead of 
> tabs? It's hard to read the pr right now as everything shows up as 
> changed.
> 
> I haven't looked at this project in quite a while - are you finding it 
> useful?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/rattigan/sonos-java-api/pull/5#issuecomment-168100088.
